pkg/variables/parsers/VariableTemplateParser.go

```diff
@@ -21,18 +22,102 @@ type VariableTemplateParser interface {
 	ExtractVariables(template string) ([]string, error)
 	//ParseTemplate(template string, values map[string]string) string
 	ParseTemplate(parserRequest VariableParserRequest) VariableParserResponse
+	//ParseTemplateWithPrimitiveHandling(parserRequest VariableParserRequest) VariableParserResponse
 }
 
 type VariableTemplateParserImpl struct {
-	logger *zap.SugaredLogger
+	logger                       *zap.SugaredLogger
+	VariableTemplateParserConfig *VariableTemplateParserConfig
 }
 
-func NewVariableTemplateParserImpl(logger *zap.SugaredLogger) *VariableTemplateParserImpl {
-	return &VariableTemplateParserImpl{logger: logger}
+func NewVariableTemplateParserImpl(logger *zap.SugaredLogger) (*VariableTemplateParserImpl, error) {
+	impl := &VariableTemplateParserImpl{logger: logger}
+	cfg, err := GetVariableTemplateParserConfig()
+	if err != nil {
+		return nil, err
+	}
+	impl.VariableTemplateParserConfig = cfg
+	return impl, nil
+}
+
+type VariableTemplateParserConfig struct {
+	ScopedVariableEnabled          bool `env:"SCOPED_VARIABLE_ENABLED" envDefault:"false"`
+	ScopedVariableHandlePrimitives bool `env:"SCOPED_VARIABLE_HANDLE_PRIMITIVES" envDefault:"false"`
+}
+
+func GetVariableTemplateParserConfig() (*VariableTemplateParserConfig, error) {
+	cfg := &VariableTemplateParserConfig{}
+	err := env.Parse(cfg)
+	return cfg, err
+}
+
+func preProcessPlaceholder(template string) string {
+	pattern := `\"@{{([^}]+)}}\"`
```

Comment:
make a constant of this regex and why aren't we using our variable regex here ??

Reply:
have used the same regex now but since I'm using this regex to extract inner content I the regex I need is slightly different.